### DISCUSSION TITLE
Confusing `OnyxNavButton` API

### DISCUSSION THREAD

**Discussion by bergser** [SchwarzIT/onyx#2112] [Q&A]
[OnyxNavButton](https://github.com/SchwarzIT/onyx/blob/main/packages/sit-onyx/src/components/OnyxNavBar/modules/OnyxNavButton/OnyxNavButton.vue) has a property `href`, which is used only for emitting `navigate` event on `ParentClick` if button has no children.

Usual expectation from `href` attribute is also [hyperlink ](https://www.w3schools.com/tags/tag_a.asp) behavior. At least it should 

- be rendered as `a` tag
- have `rel` and `target` attributes.
- display `href` in the browser status bar if element is hovered

Having `a` tags in NavBar is common use case. Is there any plans to change this in the future or what was the idea behind this implementation?

**Comment by JoCa96** [comment]
Hi @SergeiZheleznov, thanks a lot for the feedback! You are absolutely right, anchor/hyperlink a default use-case for the navbar. We have it in our backlog to support `button-as-link` and `link-as-button` behavior ([#1427](https://github.com/SchwarzIT/onyx/issues/1427), [#1947](https://github.com/SchwarzIT/onyx/issues/1947), [#1948](https://github.com/SchwarzIT/onyx/issues/1948)).

I understand, that we should prioritize this topic higher.

Also, we are planning to re-work parts of the navbar to make the usage more straight forward.

**Comment by JoCa96** [comment]
@bergser We reworked the navigation API  with [`beta-188`](https://onyx.schwarz/development/packages/changelogs/sit-onyx.html#_1-0-0-beta-188). `OnyxNavItem` will render as `button` by default, and as `a` with Vue-Router integration when the link prop is used. 

Check out the code snippets in the Storybook:
- https://storybook.onyx.schwarz/?path=/story/navigation-navbar--default
- https://storybook.onyx.schwarz/?path=/docs/navigation-modules-navitem--docs
